### PR TITLE
feat: auto publishing for monorepo package

### PR DESCRIPTION
**Pull request by Louis-Amas**

### CODE REVIEW

File: .github/workflows/publish-mangrove-js.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+        passphrase: ${{ secrets.PASSPHRASE }}
+    
+    - name: Yarn install
+      run: |
+        node --version
+        npm install -g yarn
+        yarn --version
+        yarn install --immutable
+
```
[the remark this answers is not in the document]
Comment (Louis-Amas) [reply .github/workflows/publish-mangrove-js.yml:47]: Effectively, we can remove npm instrall -g yarn, it was use in testing

File: .github/workflows/publish-mangrove-js.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        | awk -F: '{ print $2 }' \
+        | sed 's/[",]//g' | cut -c2- )
+        echo "version: $PACKAGE_VERSION"
+        git switch -c "release/v$PACKAGE_VERSION"
+        git add package.json 
+        git commit -m "$PACKAGE_VERSION [$USER]"
+        git tag -m "v$PACKAGE_VERSION" -s "v$PACKAGE_VERSION"
+        git push --set-upstream origin release/v$PACKAGE_VERSION
+        git push --tags
+        export RELEASE_NOTES=`sed -n "/^# $PACKAGE_VERSION/,/^#/p" CHANGELOG.md | grep -v "^#" | sed -e :a -e '/./,$!d;/^\n*$/{$d;N;};/\n$/ba'`
```
[the remark this answers is not in the document]
Comment (Louis-Amas) [reply .github/workflows/publish-mangrove-js.yml:100]: Nothing, we expect the changelog to be filled under # new version

File: .github/workflows/publish-mangrove-js.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Louis-Amas) [reply .github/workflows/publish-mangrove-js.yml:46]: Removed!

File: .github/workflows/publish-commonlib-js.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    - name: setup git
+      run: |
+        echo "INPUT_COMMIT_USER_NAME: ${INPUT_COMMIT_USER_NAME}";
+        git config --global user.name  "${INPUT_COMMIT_USER_NAME}";
+        echo "INPUT_COMMIT_USER_EMAIL: ${INPUT_COMMIT_USER_EMAIL}";
+        git config --global user.email ${INPUT_COMMIT_USER_EMAIL}
+
+      env:
+        INPUT_COMMIT_USER_NAME:  github-actions-bot
+        INPUT_COMMIT_USER_EMAIL: github-actions-bot@mangrove.exchange
```
[the remark this answers is not in the document]
Comment (Louis-Amas) [reply .github/workflows/publish-commonlib-js.yml:82]: Good catch

File: .github/workflows/publish-mangrove-js.yml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        else
+          tag="latest"
+        fi
+        echo "npmAuthToken: \"$NPM_TOKEN\"" >> ~/.yarnrc.yml
+        cd packages/mangrove.js
+
+        echo 'yarn'
+        yarn --version
+        cat ~/.yarnrc.yml
+        cat ~/.yarnrc.yml >> .yarnrc.yml
```
Comment (lnist) [inline .github/workflows/publish-mangrove-js.yml:68]: why this?

**Review by lnist — APPROVED** [review]

File: .github/workflows/publish-mangrove-js.yml (hunk 4 above)
Comment (Louis-Amas) [reply .github/workflows/publish-mangrove-js.yml:68]: This is what I call black magic

Comment (Louis-Amas) [reply .github/workflows/publish-mangrove-js.yml:68]: yarn berry is so bad that you never know which .yarnrc.yml is going to use, so making sure that they are similar is usefull, but we could try to remove it

**Pull request merged**